mantidproject/mantid
# Indirect convolutional simultaneous fits

**Description of work.**

This adds simultaneous fitting to to the convolutional fitting tab of the Data Analysis tab. This is an extensive rewrite of the tab so that it now conforms to the new design.

See https://godbolt.org/z/q59Lkw for minimal working demo of disabled clang compiler warning.

As this branch was getting overly large and has reached a semi-stable state I have decided to merge it into a feature branch now and deal  with remaining issues in new pull requests. Outstanding issues which will be dealt with elsewhere are.

  * Adding in the temperature correction.
  * Updating the Q display in full function to show the correct Q values.
  * Fix the naming in Iqt fit naming #27328 was temporarily reverted as it broke the convolutional fits.
  * Add scientific constraints to fitting functions.
  * Add the centre delta function parameter to the fit template.



**To test:**
  * Code review
  * Relevant files to test the fitting can be found in #26631
  * Go to the Convolutional fitting tab and try a few fits, particular things to check are:
     * Do the resultant workspace names look correct.
     * Does the fitting function correctly in multi-input mode.



Fixes #26881  




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
